src/Tasks/LegalProposition/Helpers.hs

```diff
@@ -1,9 +1,9 @@
 {-# LANGUAGE RecordWildCards #-}
 module Tasks.LegalProposition.Helpers where
-import Tasks.SynTree.Config (SynTreeConfig (..))
+import Tasks.SynTree.Config (SynTreeConfig (..), arrowOperatorsAllowed)
 import Trees.Types (BinOp)
 
 formulaAmount :: SynTreeConfig -> Integer
-formulaAmount SynTreeConfig{..} =
-  (if allowArrowOperators then fromIntegral (length [minBound..maxBound :: BinOp]) else 2)
+formulaAmount cfg@SynTreeConfig{..} =
+  (if arrowOperatorsAllowed cfg then fromIntegral (length [minBound..maxBound :: BinOp]) else 2)
```

Comment:
Something similar to above observation, but here probably more simple. The number we are looking in this case is probably simply the number of binary operators set to non-zero frequency.